src/geopathfinder/file_naming.py

```diff
@@ -334,10 +367,14 @@ def _build_fn(self):
             Filled file name.
 
         """
+
         fn_parts = list(self._fn_map.values())
         filename = str(fn_parts[0])
         for i in range(1, len(fn_parts)):
-            filename = filename[:-len(fn_parts[i-1])] + (fn_parts[i-1] + fn_parts[i])
+            if len(fn_parts[i-1]) == 0:
+                filename = filename + (fn_parts[i - 1] + fn_parts[i])
+            else:
+                filename = filename[:-len(fn_parts[i-1])] + (fn_parts[i-1] + fn_parts[i])
```

Comment:
That part can be made shorter, I will fix this, because this was originally implemented by myself.